repositories/inMemory.go

```diff
@@ -0,0 +1,26 @@
+package repositories
+
+import (
+	"sync/atomic"
+	"tsm/models"
+)
+
+var (
+	DeviceRepo deviceRepoInterface = &inMemoryRepository{devices: make(map[int]models.DeviceWithId)}
```

Comment:
Global, public state will get you into trouble. Pretty soon you will change inMemory repo to mongoDB based one - you do not want to create new connection when someone just loads this package (for example in tests).

It is common to provide factory/constructor function, keep dependencies inside the struct and create an instance in main (or some kind of factory, dependency provider).

My suggestion is to have:
```
func NewInMemRepo() *DeviceRepo {
....
}
```